types/uglifyjs-webpack-plugin/index.d.ts

```diff
@@ -0,0 +1,43 @@
+// Type definitions for uglifyjs-webpack-plugin 1.1
+// Project: https://github.com/webpack-contrib/uglifyjs-webpack-plugin
+// Definitions by: Rene Vajkay <https://github.com/vajkayrene>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.3
+
+import { Plugin } from 'webpack';
+
+export = UglifyJsPlugin;
+
+declare class UglifyJsPlugin extends Plugin {
+    constructor(options?: UglifyJsPlugin.UglifyJsPluginOptions);
+}
+
+declare namespace UglifyJsPlugin {
+    interface UglifyJsPluginOptions {
+        test?: RegExp | RegExp[];
+        include?: RegExp | RegExp[];
+        exclude?: RegExp | RegExp[];
+        cache?: boolean | string;
+        parallel?: boolean | number;
+        sourceMap?: boolean;
+        uglifyOptions?: UglifyJsOptions;
+        extractComments?: boolean | RegExp | ((node: object, comment: string) => boolean) | ExtractCommentsOptions;
+        warningsFilter?: (source: string) => boolean;
+    }
+
+    interface UglifyJsOptions {
```

Comment:
I think this should be `MinifyOptions`from  https://github.com/alan-agius4/DefinitelyTyped/blob/6477e6e34f2fdf7c86ddd229cc3e1d79b7bc23cc/types/uglify-js/index.d.ts

Reply:
I see your point. But the thing that bothers me is here is that there is no 1to1 mapping between MinifyOptions and UglifyJsOptions. I need to check how UglifyJsPlugin passes over UglifyJsOptions to UglifyJs.

Reply:
Since these structures aren't 1to1 copies of each other. I decided to go the safer way for now and stick to the fields from UglifyJSPlugin documentation.